docs/index.rst

```diff
@@ -101,19 +101,22 @@ Basic usage
 
 .. testcode::
 
-   class MLP(nn.Module):
+   class MLP(nn.Module):                    # dataclass Modules
+     out_dims: int                         
+
      @nn.compact
      def __call__(self, x):
-       x = nn.Dense(16)(x)                # inline submodules
-       x = nn.relu(x)
-       x = nn.Dense(16)(x)                # inline submodules
+       x = x.reshape((x.shape[0], -1))       
+       x = nn.Dense(128)(x)                 # create submodules inline
```

Comment:
Suggestion: In case some users may not have read the [`flax.linen.Module`](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html) docs - where it says "Modules can contain submodules" - then how about the following

```suggestion
       x = nn.Dense(128)(x)                 # create inline Flax Module submodules
```

Note the word order, especially for "inline".